pkg/pillar/containerd/containerd.go

```diff
@@ -706,35 +710,43 @@ func createMountPointExecEnvFiles(containerPath string, mountpoints map[string]s
 // it R/O into the container. On top of that we expect the usual suspects of /run,
 // /persist and /config to be taken care of by the OCI config that lk produced.
 func LKTaskLaunch(name, linuxkit string, domSettings *types.DomainConfig, args []string) (int, error) {
-	config := "/containers/services" + linuxkit + "/config.json"
-	rootfs := "/containers/services" + linuxkit + "/lower"
+	config := "/containers/services/" + linuxkit + "/config.json"
+	rootfs := "/containers/services/" + linuxkit + "/rootfs"
 
-	f, err := os.Open(config)
+	log.Infof("Starting LKTaskLaunch for %s", linuxkit)
+	f, err := os.Open("/hostfs" + config)
 	if err != nil {
 		return 0, fmt.Errorf("can't open spec file %s %v", config, err)
 	}
 
 	spec, err := NewOciSpec(name)
 	if err != nil {
+		log.Errorf("NewOciSpec failed with error %v", err)
 		return 0, err
 	}
 	if err = spec.Load(f); err != nil {
 		return 0, fmt.Errorf("can't load spec file from %s %v", config, err)
 	}
 
 	spec.Root.Path = rootfs
-	spec.Root.Readonly = true
 	if domSettings != nil {
```

Comment:
Same question here

Reply:
I had to make rootfs as R/W because of  /dev/, /sys, /proc systems, since qemu/kernel is creating /dev/pts/ and  /proc/ files during the VM launch.

Reply:
reverted the Readonly change to true.

Reply:
The only issue with read-only root partition is that, we can't use "screen" with default settings,
since it tries to create /tmp/.screens directory, and it fails. But we can always override that by
explicitly mentioning "SCREENDIR" env variable. So we are all good with the Readonly root fs.